Add cookbooks for package and scala scripts (#174)
VirtusLab/scala-cli: website/docs/cookbooks/scala-package.md

---

> title: Package scala application to executable file

Reviewer: ?
```suggestion
title: Package Scala application as an executable file
```

sidebar_position: 2
---

> ScalaCli allows you to package your application into `bootstrap` JAR file, that can be easily run.

Reviewer: ```suggestion
Scala CLI allows you to package your application into a lightweight JAR file, that can be easily run.
```

> It only contains the source byte code with no dependencies that must be automatically downloaded on first run.

Reviewer: ```suggestion
It only contains the byte code of your sources, and automatically downloads its dependencies on its first run.
```

> The following snippect contains short application to detecting the OS:

Reviewer 2: ```suggestion
The following snippet contains a short application to detect the OS:
```

```scala name:DetectOsApp.scala
object DetectOSApp extends App  {
    def getOperatingSystem(): String = {
        val os: String = System.getProperty("os.name")
        os
    }
    println(s"os: ${getOperatingSystem()}")
}
```

> ### Default format (Bootstrap)

Reviewer: ```suggestion
### Default format (lightweight launcher)
```

> scala-cli `package` command generate executable `DetectOsApp` bootstrap file. 

Reviewer: ```suggestion
By default, the `package` sub-command generates a lightweight JAR.
```

```scala-cli
scala-cli package DetectOsApp.scala
```

<!-- Expected:
Wrote DetectOsApp, run it with
  ./DetectOsApp
-->

> Bootstrap format require, that `java` command is available in the `PATH` and access to internet. 

Reviewer: ```suggestion
Lightweight JARs require the `java` command to be available, and access to internet if dependencies need to be downloaded.
```

```bash
# Run DetectOsApp on MacOs 
./DetectOsApp
# os: Mac OS X
```

> ```bash

Reviewer 2: It is unclear why we are showing docker here. I think we should include a sentence or two why are showing this snippets

# Using docker to run DetectOsApp on linux
docker run -it  -v $(pwd)/:/app -w /app openjdk:8-jre-slim bash
root@b0459df02263:/app# ./DetectOsApp
# os: Linux
``` 

### Assemblies

> scala-cli `package --assembly` command generate executable `assemblies` or `fat JARs` file. 

Reviewer: ```suggestion
Passing `--assembly` to the `package` sub-command generates so-called "assemblies" or "fat JARs". 
```

```scala-cli
scala-cli package --assembly DetectOsApp.scala
```

> The Assemblies format also requires that the `java` command be available in` PATH`, but in this case all dependencies are packed into file, so there is no need to automatically download anything before first run.

Reviewer: ```suggestion
Assemblies also require the `java` command to be available in the `PATH`. As all dependencies are packaged into the assembly, nothing gets downloaded upon the first run and no internet access is required.
```
